src/routes/api/assets/deleteById.js

```diff
@@ -1,16 +1,62 @@
 const logger = require("../../../logger");
+const z = require("zod");
+const { deleteAsset } = require("../../../model/asset");
 const { createSuccessResponse } = require("../../../response");
+const prisma = require("../../../model/data/prismaClient");
 
 /**
- * Delete an asset by it's id
+ * Delete an asset by its id after verifying ownership
  */
 
-// eslint-disable-next-line no-unused-vars
 module.exports = async (req, res, next) => {
   logger.debug(
     { user: req.user, id: req.params.id },
     `received request: DELETE /v1/assets/:assetId`,
   );
 
-  return res.status(418).json(createSuccessResponse());
+  const assetId = await req.params.assetId;
+
+  try {
+    z.string().uuid().parse(assetId);
+  } catch (error) {
+    logger.debug({ error }, "invalid asset ID format");
+    return next({ status: 400, message: "invalid asset id" });
+  }
+
+  let asset;
+  try {
+    asset = await prisma.asset.findUnique({
+      where: { uuid: assetId },
+      include: {
+        character: true,
+        user: true,
+      },
+    });
+  } catch (error) {
+    logger.error(error, "Error fetching asset");
+    return next({ status: 500, message: "Internal server error" });
+  }
+
+  if (!asset) {
+    logger.debug("asset not found");
+    return next({ status: 404, message: "Asset not found" });
+  }
+
+  if (asset.user.hashedEmail !== req.user) {
```

Comment:
all of this should be handled in a single Prisma query by adding the user's id to the where clause in deleteAsset in model/asset.js